### DISCUSSION TITLE
zh/datasets/detect/brain-tumor/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9988] [Docs]
# zh/datasets/detect/brain-tumor/

脑肿瘤检测是医学成像领域的一个领先数据集，与Ultralytics 集成。探索使用它训练YOLO 模型的方法。

https://docs.ultralytics.com/zh/datasets/detect/brain-tumor/

**Comment by ZoltinZyc** [comment]
how can i download this datasett??

**Reply by pderrenger** [reply]
Hey there! 😊 To download the brain tumor dataset, you can use the direct download link we provide in our documentation. Simply use the following command in your terminal to download and unzip the dataset:

```bash
wget https://ultralytics.com/assets/brain-tumor.zip
unzip brain-tumor.zip -d path/to/destination
```

Don't forget to replace `path/to/destination` with the directory where you'd like the dataset to be stored. Once you have the dataset, you're all set to start training or inferencing with YOLO models on brain tumor images!

Should you need further guidance or have more questions, feel free to ask. Happy coding! 🚀

**Comment by xySun17** [comment]
How can i train the yoloV8 using this dataset without loading the pre-trained weights?

**Reply by pderrenger** [reply]
@xySun17 hello!

To train YOLOv8 without loading the pre-trained weights, you can initialize the model from scratch by specifying the model architecture file (e.g., `yolov8n.yaml`) instead of the pre-trained weights file. Here’s how you can do it:

### Python
```python
from ultralytics import YOLO

# Load a model architecture without pre-trained weights
model = YOLO("yolov8n.yaml")  # build a new model from YAML

# Train the model
results = model.train(data="brain-tumor.yaml", epochs=100, imgsz=640)
```

### CLI
```bash
# Train from scratch using the model architecture file
yolo detect train data=brain-tumor.yaml model=yolov8n.yaml epochs=100 imgsz=640
```

This approach ensures that the model starts training with random weights rather than pre-trained ones. If you encounter any issues, please ensure you are using the latest version of the Ultralytics package and provide a minimum reproducible example if the problem persists. You can find more details on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to reach out if you have any further questions. Happy training! 🚀

**Reply by xySun17** [reply]
Thank you!! 
However, there is another problem, that is, why it says the path of dataset is not found? I already downloaded the dataset files and changed the path in "brain-tumor.yaml" file.

The threw error was follows:
"Dataset 'brain-tumor.yaml' images not found ⚠️, missing path '/valid/images'
Note dataset download directory is '/home/ros/datasets'. You can update this in '/home/ros/.config/Ultralytics/settings.yaml'"

**Reply by pderrenger** [reply]
Hello @xySun17,

Thank you for reaching out! It looks like the issue might be related to the dataset path configuration in your `brain-tumor.yaml` file. Here are a few steps to troubleshoot and resolve this:

1. **Verify Dataset Path**: Ensure that the paths specified in your `brain-tumor.yaml` file correctly point to the locations of your dataset images. For example:
    ```yaml
    path: /home/ros/datasets/brain-tumor
    train: train/images
    val: valid/images
    ```

2. **Check Directory Structure**: Confirm that the directory structure matches what is specified in the YAML file. The `train/images` and `valid/images` directories should exist within `/home/ros/datasets/brain-tumor`.

3. **Update Configuration**: If the dataset is located in a different directory, update the `path` in the YAML file accordingly. For instance, if your dataset is in `/home/ros/my-datasets/brain-tumor`, update the `path` to:
    ```yaml
    path: /home/ros/my-datasets/brain-tumor
    ```

4. **Reproducible Example**: If the issue persists, providing a minimum reproducible example can greatly help in diagnosing the problem. You can find guidance on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

5. **Latest Version**: Ensure you are using the latest version of the Ultralytics package. You can update it using:
    ```bash
    pip install --upgrade ultralytics
    ```

Here’s an example of how your `brain-tumor.yaml` should look:
```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# Brain-tumor dataset by Ultralytics
# Documentation: https://docs.ultralytics.com/datasets/detect/brain-tumor/
# Example usage: yolo train data=brain-tumor.yaml

path: /home/ros/datasets/brain-tumor  # dataset root dir
train: train/images  # train images (relative to 'path')
val: valid/images  # val images (relative to 'path')
test:  # test images (relative to 'path')

# Classes
names:
  0: negative
  1: positive

# Download script/URL (optional)
download: https://github.com/ultralytics/assets/releases/download/v0.0.0/brain-tumor.zip
```

If you follow these steps and still encounter issues, please share more details or any error messages you receive. This will help us assist you better.

Best regards and happy coding! 🚀

**Comment by pizzas17** [comment]
How could I download this dataset on COCO format or how could I convert it? (Since I wish to have an EDA on it

**Comment by G-qingqian** [comment]
Can your dataset ensure that the data are independently and identically distributed by patient? That is, there will be no data from the same patient in both the training set and the validation set.

**Reply by glenn-jocher** [reply]
The packaged `brain-tumor` dataset split is provided at the image-folder level and doesn’t ship patient IDs/metadata, so we can’t guarantee a patient-disjoint (no same patient in both train/val) split; if you need patient-wise separation, you’ll need to rebuild the split from patient identifiers and point a custom YAML to `*.txt` image lists (supported by Ultralytics YOLO), e.g. see the provided config in the [Brain Tumor dataset docs](https://docs.ultralytics.com/datasets/detect/brain-tumor/) and the [brain-tumor.yaml in the repo](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/brain-tumor.yaml).

```yaml
# my-brain-tumor.yaml
path: /data/brain-tumor
train: splits/train.txt  # one image path per line (ideally grouped by patient ID when you create it)
val: splits/val.txt
names:
  0: negative
  1: positive
```

**Comment by G-qingqian** [comment]
Thank you for your reply. I noticed that in the original dataset, the images in the train folder do have ID prefixes, but the images in the val folder are all prefixed with val1_, making it impossible to distinguish which patient the data belongs to. Therefore, I cannot redivide the dataset. Could you please provide data with complete identifiers? 
---- Replied Message ----
FromGlenn ***@***.***>Date1/18/2026 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [ultralytics/ultralytics] zh/datasets/detect/brain-tumor/ (Discussion #9988)
The packaged brain-tumor dataset split is provided at the image-folder level and doesn’t ship patient IDs/metadata, so we can’t guarantee a patient-disjoint (no same patient in both train/val) split; if you need patient-wise separation, you’ll need to rebuild the split from patient identifiers and point a custom YAML to *.txt image lists (supported by Ultralytics YOLO), e.g. see the provided config in the Brain Tumor dataset docs and the brain-tumor.yaml in the repo.
# my-brain-tumor.yaml path: /data/brain-tumor train: splits/train.txt # one image path per line (ideally grouped by patient ID when you create it) val: splits/val.txt names: 0: negative 1: positive

**Reply by glenn-jocher** [reply]
The packaged `brain-tumor` zip only includes image-level labels with anonymized filenames (no patient IDs/metadata), so we can’t provide “complete identifiers” or guarantee a patient-disjoint re-split; for patient-wise splitting you’ll need to rebuild from a source that includes patient IDs (e.g., your DICOM/NIfTI originals) and then generate `train.txt`/`val.txt`, and if you just need stronger evaluation without IDs you can use our [K-Fold Cross Validation guide](https://docs.ultralytics.com/guides/kfold-cross-validation/)—please keep follow-ups here in the GitHub discussion (we can’t provide support over email/DM).
